scripts/install_dependencies/install_openvino_dependencies.sh

```diff
@@ -16,260 +16,341 @@
-    echo "WARNING: This repository is NOT PROVIDED OR SUPPORTED by Intel or CentOS. Neither Intel nor CentOS has control over this repository. Terms governing your use of FFmpeg can be found here: https://www.ffmpeg.org/legal.html "
-    echo "Once added, this repository will be enabled on your operating system and can thus receive updates to all packages installed from it. "
-    echo
-    echo "Consider the following ways to prevent unintended 'updates' from this third party repository from over-writing some core part of CentOS:"
-    echo "a) Only enable these archives from time to time, and generally leave them disabled. See: man yum"
-    echo "b) Use the exclude= and includepkgs= options on a per sub-archive basis, in the matching .conf file found in /etc/yum.repos.d/ See: man yum.conf"
-    echo "c) The yum Priorities plug-in can prevent a 3rd party repository from replacing base packages, or prevent base/updates from replacing a 3rd party package."
-    echo
-    echo "Option 2: Skip FFmpeg installation."
-    echo
-
-    if yes_or_no_ffmpeg; then
-        if [[ -z $RPMFUSION_IS_INSTALLED ]]; then
-            yum install -y epel-release
-            rpm -Uvh https://download1.rpmfusion.org/free/el/rpmfusion-free-release-7.noarch.rpm
-        fi
-        yum install -y ffmpeg
+fi
+
+#===================================================================================================
+# Gather packages and print list
+
+pkgs=()
+for comp in ${comp[@]} ; do
+    var=pkgs_${comp}[@]
+    pkgs+=(${!var})
+done
+
+if [ ${#pkgs[@]} -eq 0 ]; then
+    if  [ -n "$print" ] ; then
+        echo "No packages to install" >&2
+        exit 1
     else
-        echo "FFmpeg installation skipped. You may build FFmpeg from sources as described here: https://trac.ffmpeg.org/wiki/CompilationGuide/Centos"
-        echo
+        echo "No packages to install"
+        exit 0
     fi
-    exit
 fi
+
+if  [ -n "$print" ] ; then
+    echo "${pkgs[*]}"
+    exit 0
+fi
+
+#===================================================================================================
+# Actual installation
+
+if [ $EUID -ne 0 ]; then
+    echo "ERROR: this script must be run as root to install 3rd party packages." >&2
+    echo "Please try again with \"sudo -E $0\", or as root." >&2
+    exit 1
+fi
+
+iopt=
+
+if [ "$os" == "ubuntu18.04" ] || [ "$os" == "ubuntu20.04" ] ; then
+
+    [ -z "$interactive" ] && iopt="-y"
+    [ -n "$dry" ] && iopt="--dry-run"
+    [ -n "$keepcache" ] && rm -f /etc/apt/apt.conf.d/docker-clean
+
+    apt-get update && apt-get install --no-install-recommends $iopt ${pkgs[@]}
+
+elif [ "$os" == "centos7" ] ; then
+
+    [ -z "$interactive" ] && iopt="--assumeyes"
+    [ -n "$dry" ] && iopt="--downloadonly"
+    [ -n "$keepcache" ] && iopt="$iopt --setopt=keepcache=1"
+    [ ${#extra_repos[@]} -ne 0 ] && yum localinstall $iopt --nogpgcheck ${extra_repos[@]}
+
+    yum install $iopt ${pkgs[@]}
+
+fi
```

Comment:
It make sense to add `else` branch with error message (to reduce mistakes in the future).

(similar note for line 300)

Reply:
Done, thank you!